pkg/deploy/migration/on-reconcile-one-time-migration.go

```diff
@@ -0,0 +1,287 @@
+		}
+		if err := addRequiredLabelsForSecret(ctx, tlsSecretName); err != nil {
+			return err
+		}
+	}
+
+	// TLS
+	if err := addRequiredLabelsForSecret(ctx, deploy.CheTLSSelfSignedCertificateSecretName); err != nil {
+		return err
+	}
+
+	if ctx.CheCluster.Spec.Server.CheHostTLSSecret != "" {
+		if err := addRequiredLabelsForSecret(ctx, ctx.CheCluster.Spec.Server.CheHostTLSSecret); err != nil {
+			return err
+		}
+	}
+
+	// Proxy credentials
+	if ctx.CheCluster.Spec.Server.ProxySecret != "" {
+		if err := addRequiredLabelsForSecret(ctx, ctx.CheCluster.Spec.Server.ProxySecret); err != nil {
+			return err
+		}
+	}
+
+	// Database credentials
+	if ctx.CheCluster.Spec.Database.ChePostgresSecret != "" {
+		if err := addRequiredLabelsForSecret(ctx, ctx.CheCluster.Spec.Database.ChePostgresSecret); err != nil {
+			return err
+		}
+	}
+
+	// Keycloak related secrets
+	if ctx.CheCluster.Spec.Auth.IdentityProviderSecret != "" {
+		if err := addRequiredLabelsForSecret(ctx, ctx.CheCluster.Spec.Auth.IdentityProviderSecret); err != nil {
+			return err
+		}
+	}
+
+	if ctx.CheCluster.Spec.Auth.IdentityProviderPostgresSecret != "" {
+		if err := addRequiredLabelsForSecret(ctx, ctx.CheCluster.Spec.Auth.IdentityProviderPostgresSecret); err != nil {
+			return err
+		}
+	}
+
+	// OAuth
+	if ctx.CheCluster.Spec.Auth.OAuthSecret != "" {
+		if err := addRequiredLabelsForSecret(ctx, ctx.CheCluster.Spec.Auth.OAuthSecret); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func addRequiredLabelsForConfigMap(ctx *deploy.DeployContext, configMapName string) error {
+	configMap := &corev1.ConfigMap{}
+	return addRequiredLabelsForObject(ctx, configMapName, configMap)
+}
+
+func addRequiredLabelsForSecret(ctx *deploy.DeployContext, secretName string) error {
+	secret := &corev1.Secret{}
+	return addRequiredLabelsForObject(ctx, secretName, secret)
+}
+
+// addRequiredLabelsForObject adds standard set of labels to the object with given name
+// As the function doesn't know the kind of the object with given name an empty object should be passed,
+// for example: addRequiredLabelsForObject(ctx, "my-secret", &corev1.Secret{})
+func addRequiredLabelsForObject(ctx *deploy.DeployContext, objectName string, obj client.Object) error {
+	// Check if the object is already migrated
+	if exists, _ := deploy.GetNamespacedObject(ctx, objectName, obj); exists {
+		// Default client sees the object in cache, no need in adding labels
+		return nil
+	}
+
+	err := ctx.ClusterAPI.NonCachingClient.Get(context.TODO(), types.NamespacedName{Namespace: ctx.CheCluster.Namespace, Name: objectName}, obj)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// The object doesn't exist in cluster, nothing to do
+			return nil
+		}
+		return err
+	}
+
+	// Add required labels to the object
+	labels := obj.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	for labelName, labelValue := range deploy.GetLabels(ctx.CheCluster, deploy.DefaultCheFlavor(ctx.CheCluster)) {
```

Comment:
Set just one needed label. don't set all of them.

Reply:
ok